scripts/manageacl.py

```diff
@@ -143,14 +143,16 @@ async def enable_acls(self, endpoint: str):
             await search_index_client.create_or_update_index(index_definition)
 
 
-async def main(args: any):
+async def main(args: Any):
     # Use the current user identity to connect to Azure services unless a key is explicitly set for any of them
     azd_credential = (
         AzureDeveloperCliCredential()
         if args.tenant_id is None
         else AzureDeveloperCliCredential(tenant_id=args.tenant_id, process_timeout=60)
     )
-    search_credential = azd_credential if args.search_key is None else AzureKeyCredential(args.search_key)
+    search_credential: Union[AsyncTokenCredential, AzureKeyCredential] = azd_credential
```

Comment:
To make mypy happy, I had to break the ternary into an if/else. It's a known issue with ternaries:
https://github.com/python/mypy/issues?q=is%3Aissue+is%3Aopen+ternary